collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -206,6 +233,91 @@ func (r *telemetryAPIReceiver) getRecordRequestId(record map[string]interface{})
 	return ""
 }
 
+func (r *telemetryAPIReceiver) createMetrics(slice []event) (pmetric.Metrics, error) {
+	metric := pmetric.NewMetrics()
+	resourceMetric := metric.ResourceMetrics().AppendEmpty()
+	r.resource.CopyTo(resourceMetric.Resource())
+	scopeMetric := resourceMetric.ScopeMetrics().AppendEmpty()
+	scopeMetric.Scope().SetName(scopeName)
+	scopeMetric.SetSchemaUrl(semconv.SchemaURL)
+
+	for _, el := range slice {
+		r.logger.Debug(fmt.Sprintf("Event: %s", el.Type), zap.Any("event", el))
+		record, ok := el.Record.(map[string]any)
+		if !ok {
+			continue
+		}
+		ts, err := time.Parse(time.RFC3339, el.Time)
+		if err != nil {
+			continue
+		}
+
+		switch el.Type {
+		case string(telemetryapi.PlatformInitStart):
+			r.faaSMetricBuilders.coldstartsMetric.Add(1)
+			r.faaSMetricBuilders.coldstartsMetric.AppendDataPoints(scopeMetric, pcommon.NewTimestampFromTime(ts))
+		case string(telemetryapi.PlatformInitReport):
+			metrics, ok := record["metrics"].(map[string]any)
+			if !ok {
+				continue
+			}
+
+			status, _ := metrics["status"].(string)
```

Comment:
Doesn't look like the record's `metrics` field has a status field: https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#platform-initReport

```suggestion
			metrics, ok := record["metrics"].(map[string]any)
			if !ok {
				continue
			}

			status, _ := record["status"].(string)
```

Reply:
Updated, should have been on the main `record` object.